intel_extension_for_transformers/langchain/embeddings/optimized_sentence_transformers.py

```diff
@@ -53,12 +54,84 @@ def __init__(self, *args, **kwargs):
+    def _load_auto_model(
+            self, model_name_or_path: str, token: Optional[Union[bool, str]], cache_folder: Optional[str]):
         """
         Creates a simple Transformer + Mean Pooling model and returns the modules
         """
         logger.warning("No sentence-transformers model found with name {}." \
                        "Creating a new one with MEAN pooling.".format(model_name_or_path))
-        transformer_model = OptimzedTransformer(model_name_or_path, cache_dir=cache_folder, model_args={"token": token})
-        pooling_model = sentence_transformers.models.Pooling(transformer_model.get_word_embedding_dimension(), 'mean')
-        return [transformer_model, pooling_model]
\ No newline at end of file
+        transformer_model = OptimzedTransformer(
+            model_name_or_path, cache_dir=cache_folder, model_args={"token": token})
+        pooling_model = sentence_transformers.models.Pooling(
+            transformer_model.get_word_embedding_dimension(), 'mean')
+        return [transformer_model, pooling_model]
+    
+    def _load_sbert_model(
+            self, model_name_or_path: str, token: Optional[Union[bool, str]], cache_folder: Optional[str]):
+        """
+        Loads a full sentence-transformers model
+        """
+        # Check if the config_sentence_transformers.json file exists (exists since v2 of the framework)
+        config_sentence_transformers_json_path = sentence_transformers.util.load_file_path(
+            model_name_or_path, 'config_sentence_transformers.json', token=token, cache_folder=cache_folder)
+        if config_sentence_transformers_json_path is not None:
+            with open(config_sentence_transformers_json_path) as fIn:
+                self._model_config = json.load(fIn)
+
+            if '__version__' in self._model_config and \
+                'sentence_transformers' in self._model_config['__version__'] and \
+                self._model_config['__version__']['sentence_transformers'] > sentence_transformers.__version__:
+                logger.warning("You try to use a model that was created with version {}, "\
+                               "however, your version is {}. This might cause unexpected "\
+                               "behavior or errors. In that case, try to update to the "\
+                               "latest version.\n\n\n".format(
+                                    self._model_config['__version__']['sentence_transformers'],
+                                    sentence_transformers.__version__))
+
+        # Check if a readme exists
+        model_card_path = sentence_transformers.util.load_file_path(
+            model_name_or_path, 'README.md', token=token, cache_folder=cache_folder)
+        if model_card_path is not None:
+            try:
+                with open(model_card_path, encoding='utf8') as fIn:
+                    self._model_card_text = fIn.read()

```

Comment:
Do you need _model_card_text?

Reply:
The logic for handling README file exists in the parent class's original [implementation](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/SentenceTransformer.py#L831-L838).
Even if the README file does not exist, the raised exception will be passed without causing any issues.